### DISCUSSION TITLE
Question about regular expression matching

### DISCUSSION THREAD

**Discussion by Quibusque** [draivin/hsnips#88] [Q&A] [answered]
Hi everyone,
I was trying to get a snippet for latex that turns something like "phat" into "\widehat{p}". I was able to write something that looks like this (inspired by a similar snippet I used in Ultisnips):

snippet \`([a-zA-Z])hat\` "hat" Ai
\widehat{\``rv = m[0]\``}
endsnippet

however, it looks like m[0] refers to the whole "phat" string and not just "p". I came up with something like this:

snippet \`([a-zA-Z])hat\` "hat" Ai
\widehat{\``rv = m[0].replace('hat','')\``}
endsnippet

which does get the job done but I feel like it's a rough workaround and wanted to ask if there's a simpler intended approach. Hopefully my question was clear enough.

**Comment by yfzhao20** [comment, marked as answer]
Try `rv=m[1]`. 
`m[0]` gives the whole regex while `m[1]` gives the first group.

**Reply by Quibusque** [reply]
Ok I guess this was a bit of a dumb question, thank you very much for helping me
